joblib/parallel.py

```diff
@@ -198,60 +354,36 @@ class parallel_backend(object):
     """
     def __init__(self, backend, n_jobs=-1, inner_max_num_threads=None,
                  **backend_params):
-        if isinstance(backend, str):
-            if backend not in BACKENDS:
-                if backend in EXTERNAL_BACKENDS:
-                    register = EXTERNAL_BACKENDS[backend]
-                    register()
-                elif backend in MAYBE_AVAILABLE_BACKENDS:
-                    warnings.warn(
-                        f"joblib backend '{backend}' is not available on "
-                        f"your system, falling back to {DEFAULT_BACKEND}.",
-                        UserWarning,
-                        stacklevel=2)
-                    BACKENDS[backend] = BACKENDS[DEFAULT_BACKEND]
-                else:
-                    raise ValueError("Invalid backend: %s, expected one of %r"
-                                     % (backend, sorted(BACKENDS.keys())))
-
-            backend = BACKENDS[backend](**backend_params)
-
-        if inner_max_num_threads is not None:
-            msg = ("{} does not accept setting the inner_max_num_threads "
-                   "argument.".format(backend.__class__.__name__))
-            assert backend.supports_inner_max_num_threads, msg
-            backend.inner_max_num_threads = inner_max_num_threads
 
-        # If the nesting_level of the backend is not set previously, use the
-        # nesting level from the previous active_backend to set it
-        current_backend_and_jobs = getattr(_backend, 'backend_and_jobs', None)
-        if backend.nesting_level is None:
-            if current_backend_and_jobs is None:
-                nesting_level = 0
-            else:
-                nesting_level = current_backend_and_jobs[0].nesting_level
-
-            backend.nesting_level = nesting_level
-
-        # Save the backends info and set the active backend
-        self.old_backend_and_jobs = current_backend_and_jobs
-        self.new_backend_and_jobs = (backend, n_jobs)
+        # warnings.warn(
+        #     "The parallel_config context manager should be used instead of "
+        #     "parallel_backend. The parallel_backend context manager might "
+        #     "be removed in future versions of joblib.",
+        #     FutureWarning,
+        # )
```

Comment:
As you pointed out in a discussion, this can be removed?

Reply:
already did :)